pypose/lietensor/lietensor.py

```diff
@@ -1220,3 +1220,32 @@ def __deepcopy__(self, memo):
             result = type(self)(self.clone(memory_format=torch.preserve_format))
             memo[id(self)] = result
             return result
+
+
+class WrappableLT:
+    tmp_md = torch.autograd.forward_ad.make_dual
+    tmp_w = torch._functorch.eager_transforms._wrap_tensor_for_grad
+    def __enter__(self):
+        torch.autograd.forward_ad.make_dual = self.make_dual_wrapper
+        torch._functorch.eager_transforms._wrap_tensor_for_grad = self._wrap_tensor_for_grad
+    
+    @classmethod
+    def make_dual_wrapper(cls, tensor, tangent, *args, level=None):
+        ltype = tensor.ltype if isinstance(tensor, LieTensor) else None
+        res = WrappableLT.tmp_md.__call__(tensor, tangent, *args, level=level)
+        if ltype is not None:
+            res = torch.Tensor.as_subclass(res, LieTensor)
+            res.ltype = ltype
```

Comment:
nit: maybe directly using LieTensor's constructor is cleaner

Reply:
as_subclass is the official recommended way, if LieTensor constructor did not call `as_subclass `